jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/contrastrules/library/LibraryVulnerabilityAssessmentPolicy.java

```diff
@@ -0,0 +1,139 @@
+        String vulnerabilities;
+        String severity = policyParam.get("severityToCheck");
+        List<VulnerabilityInfo> vulnerabilityDetailsList = new ArrayList<>();
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            vulnerabilities = resourceAttributes.get("vulns");
+            if (vulnerabilities == null || vulnerabilities.isEmpty()) {
+                return new PolicyResult(PacmanSdkConstants.STATUS_SUCCESS, PacmanRuleConstants.SUCCESS_MESSAGE);
+            }
+            ResourceVulnerability[] vulnerabilityList = mapper.readValue(vulnerabilities,
+                    new TypeReference<ResourceVulnerability[]>() {
+                    });
+            List<ResourceVulnerability> filteredList = Arrays.stream(vulnerabilityList).filter(vuln -> vuln.getSeverityToUse()
+                    .equalsIgnoreCase(severity)).collect(Collectors.toList());
+            if (filteredList.isEmpty()) {
+                return new PolicyResult(PacmanSdkConstants.STATUS_SUCCESS, PacmanRuleConstants.SUCCESS_MESSAGE);
+            }
+            String vulnerabilityUrl = String.format(CONTRAST_URL_TEMPLATE, accountId,
+                    resourceAttributes.get("fileName"));
+            filteredList.forEach(vul -> {
+                List<CveDetails> cveList = new ArrayList<>();
+                VulnerabilityInfo vulnerability = new VulnerabilityInfo();
+                vulnerability.setVulnerabilityUrl(vulnerabilityUrl);
+                vulnerability.setTitle(vul.getDescription().replace("\u00a0",""));
+                CveDetails cve = new CveDetails();
+                cve.setId(vul.getName());
+                cve.setUrl(PacmanUtils.NIST_VULN_DETAILS_URL + vul.getName());
+                cveList.add(cve);
+                vulnerability.setCveList(cveList);
+                vulnerabilityDetailsList.add(vulnerability);
+            });
+            return buildFailureAnnotation(policyParam, vulnerabilityDetailsList,
+                    resourceAttributes);
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
+            return buildFailureAnnotation(policyParam, vulnerabilityDetailsList, resourceAttributes);
+        }
+    }
+
+    @Override
+    public String getHelpText() {
+        return "Checks for the presence of vulnerabilities.";
+    }
+
+    private PolicyResult buildFailureAnnotation(final Map<String, String> policyParam,
+                                                List<VulnerabilityInfo> vulnerabilityDetailsList,
+                                                Map<String, String> resourceAttributes) {
+        ObjectMapper mapper = new ObjectMapper();
+        String violationDetails = "";
+        String fileName = resourceAttributes.get("fileName");
+        try {
+            violationDetails = mapper.writeValueAsString(vulnerabilityDetailsList);
+        } catch (Exception e) {
+            logger.error("error while converting violation details for {}", fileName);
```

Comment:
Capital "E"

Reply:
you are getting exception and not using it in any way. Why don't just log exact error? This way you won't need to add a message even.

Reply:
Updated